mitodl/micromasters
# Exams: set number of attempts to 1

#### What are the relevant tickets?
Fix #4646

#### What's this PR do?
Limit learners to one exam attempt for one payment

There are two dates to set up on the program itself `exam_attempts_first_date` and `exam_attempts_second_date`. First date is one after which all users who pay for the course after that date get 1 attempt.
one attempt.  The second date is for switching completely to 1 attempt for everybody independently of the payment date.

#### How should this be manually tested?
Set up a user with a past paid course and a proctored exam grade. Set up a new exam run that is current. The dashboard should ask the user to pay again.
If now is in between the two dates `exam_attempts_first_date` and `exam_attempts_second_date`, the learner gets 2 attempts for payment before the first date and 1 attempts after the first date. 
If now is after the `exam_attempts_second_date` then all users get 1 attempt for any payment.
